internal/provider/data_source_http.go

```diff
@@ -246,17 +247,17 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 
 	respHeadersState := types.Map{}
 
-	diags = tfsdk.ValueFrom(ctx, responseHeaders, types.Map{ElemType: types.StringType}.Type(ctx), &respHeadersState)
+	diags = tfsdk.ValueFrom(ctx, responseHeaders, types.MapNull(types.StringType).Type(ctx), &respHeadersState)
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
```

Comment:
Nit: This likely can be simplified with `respHeadersState, diags := types.MapValueFrom(ctx, types.StringType, responseHeaders)` 👍